modules/arm_plugin/src/transformations/arm_optimizations.cpp

```diff
@@ -143,6 +144,7 @@ bool ArmPlugin::pass::ArmOptimizations::run_on_function(std::shared_ptr<ngraph::
     manager.register_pass<ngraph::pass::ConvertBroadcastToTiles>();
     manager.register_pass<pass::ConvertTile>();
     manager.register_pass<pass::ConvertConcat>();
+    manager.register_pass<pass::AlignNodePrecision>();
```

Comment:
should not we call `passManager.register_pass<ngraph::pass::ConvertPrecision>(PREC, ngraph::element::f32);` for all unsupported `PREC` ? In this case all conversions are done on LoadNetwork stage, not in runtime.

Reply:
For this case, it is enough:
```cpp
#ifndef __ARM_FEATURE_FP16_VECTOR_ARITHMETIC
    manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::f16, ngraph::element::f32);
#endif
```
This transformation is needed to convert FP32 to FP16.

Reply:
```
u8 param -> relu -> u8 result
```

How will it work on ARM? We don't convert U8 for FP32 or FP16 (for Apple M1).

My expectation that we need to convert everything to FP32 for usual ARM, to FP16 for Apple M1 iike (who supports FP16 compute)

CC @apankratovantonp how do you think?

Reply:
Why don't we want to run FP32 tests on an Apple M1?

Reply:
We want, but convert FP32 to FP16 and use FP16 kernels.

If you disable conversion FP32 -> FP16 for Apple M1, you will get errors..

Reply:
We should avoid implicit accuracy reduction as it can leads to unpredictable accuracy issues. What errors we will get?

Reply:
Ok, looks like you are right.. Both FP32 and FP16 kernels are available on Apple M1. So, we can execute natively both cases